pkg/dao/rpm.go

```diff
@@ -170,3 +195,173 @@ func (r rpmDaoImpl) Search(orgID string, request api.SearchRpmRequest, limit int
 
 	return dataResponse, nil
 }
+
+// PagedRpmInsert insert all passed in rpms quickly, ignoring any duplicates
+// Returns count of new packages inserted, and any errors
+func (r rpmDaoImpl) PagedRpmInsert(pkgs *[]models.Rpm) (int64, error) {
+	var count int64
+	chunk := r.pagedRpmInsertsLimit
+	var result *gorm.DB
+	if len(*pkgs) == 0 {
+		return 0, nil
+	}
+
+	for i := 0; i < len(*pkgs); i += chunk {
+		end := i + chunk
+		if i+chunk > len(*pkgs) {
+			end = len(*pkgs)
+		}
+		result = r.db.Debug().Clauses(clause.OnConflict{
+			Columns:   []clause.Column{{Name: "checksum"}},
+			DoNothing: true,
+		}).Create((*pkgs)[i:end])
+
+		if result.Error != nil {
+			return count, result.Error
+		}
+		count += result.RowsAffected
+	}
+	return count, result.Error
+}
+
+// existingChecksums passing in a list of checksums, returns the list of checksums
+// that exist in the db
+func (r rpmDaoImpl) existingChecksums(checksums []string) (*gorm.DB, []string) {
+	var found []string
+	result := r.db.Where("checksum in ?", checksums).Model(&models.Rpm{}).Pluck("checksum", &found)
+	return result, found
+}
+
+func (r rpmDaoImpl) fetchRepo(uuid string) (models.Repository, error) {
+	found := models.Repository{}
+	result := r.db.
+		Where("UUID = ?", uuid).
+		First(&found)
+
+	if result.Error != nil {
+		return found, result.Error
+	}
+	return found, nil
+}
+
+// InsertForRepository inserts a set of yum packages for a given repository
+//   and removes any that are not in the list.  This will involve inserting the RPMs
+//   if not present, and adding or removing any associations to the Repository
+//   Returns a count of new RPMs added to the system (not the repo), as well as any error
+func (r rpmDaoImpl) InsertForRepository(repoUuid string, pkgs []yum.Package) (int64, error) {
+	var rowsAffected int64
+
+	repo, err := r.fetchRepo(repoUuid)
+	if err != nil {
+		return rowsAffected, err
```

Comment:
```suggestion
		return rowsAffected, fmt.Errorf("failed to fetch repository %w: ", err)
```

If we try to do this in more places, it's easier to debug, because then the error message is something like "failed to introspect: failed to fetch repo: record not found"